src/bicep/modules/remote-access.bicep

```diff
@@ -48,99 +51,104 @@ param windowsVmSku string
+    }
   }
-}
 
-module windowsNetworkInterface '../modules/network-interface.bicep' = {
-  name: 'remoteAccess-windowsNetworkInterface'
-  scope: resourceGroup(hubProperties.subscriptionId, hubProperties.resourceGroupName)
-  params: {
-    ipConfigurationName: hubProperties.windowsNetworkInterfaceIpConfigurationName
-    location: location
-    mlzTags: mlzTags
-    name: hubProperties.windowsNetworkInterfaceName
-    networkSecurityGroupId: hubNetworkSecurityGroupResourceId
-    privateIPAddressAllocationMethod: windowsNetworkInterfacePrivateIPAddressAllocationMethod
-    subnetId: hubSubnetResourceId
-    tags: tags
+module windowsNetworkInterface '../modules/network-interface.bicep' =
+  if (deployWindowsVirtualMachine) {
+    name: 'remoteAccess-windowsNetworkInterface'
+    scope: resourceGroup(hubProperties.subscriptionId, hubProperties.resourceGroupName)
+    params: {
+      ipConfigurationName: hubProperties.windowsNetworkInterfaceIpConfigurationName
+      location: location
+      mlzTags: mlzTags
+      name: hubProperties.windowsNetworkInterfaceName
+      networkSecurityGroupId: hubNetworkSecurityGroupResourceId
+      privateIPAddressAllocationMethod: windowsNetworkInterfacePrivateIPAddressAllocationMethod
+      subnetId: hubSubnetResourceId
+      tags: tags
+    }
   }
-}
 
-module windowsVirtualMachine '../modules/windows-virtual-machine.bicep' = {
-  name: 'remoteAccess-windowsVirtualMachine'
-  scope: resourceGroup(hubProperties.subscriptionId, hubProperties.resourceGroupName)
-  params: {
-    adminPassword: windowsVmAdminPassword
-    adminUsername: windowsVmAdminUsername
-    createOption: windowsVmCreateOption
-    diskEncryptionSetResourceId: diskEncryptionSetResourceId
-    diskName: hubProperties.windowsDiskName
-    hybridUseBenefit: hybridUseBenefit
-    location: location
-    logAnalyticsWorkspaceId: logAnalyticsWorkspaceId
-    mlzTags: mlzTags
-    name: hubProperties.windowsVmName
-    networkInterfaceName: windowsNetworkInterface.outputs.name
-    offer: windowsVmOffer
-    publisher: windowsVmPublisher
-    size: windowsVmSize
-    sku: windowsVmSku
-    storageAccountType: windowsVmStorageAccountType
-    tags: tags
-    version: windowsVmVersion
+module windowsVirtualMachine '../modules/windows-virtual-machine.bicep' =
+  if (deployWindowsVirtualMachine) {
+    name: 'remoteAccess-windowsVirtualMachine'
+    scope: resourceGroup(hubProperties.subscriptionId, hubProperties.resourceGroupName)
+    params: {
+      adminPassword: windowsVmAdminPassword
+      adminUsername: windowsVmAdminUsername
+      createOption: windowsVmCreateOption
+      diskEncryptionSetResourceId: diskEncryptionSetResourceId
+      diskName: hubProperties.windowsDiskName
+      hybridUseBenefit: hybridUseBenefit
+      location: location
+      logAnalyticsWorkspaceId: logAnalyticsWorkspaceId
+      mlzTags: mlzTags
+      name: hubProperties.windowsVmName
+      networkInterfaceName: windowsNetworkInterface.outputs.name
```

Comment:
@tsweatman1 line 145 needs a condition to fix your deployment error, like so:
networkInterfaceName: deployWindowsVirtualMachine ? windowsNetworkInterface.outputs.name : ''

Reply:
Added the necessary conditions and tested with and without Bastion.  Linux VM still needs omsextension, otherwise Option to choose which VMs to deploy has been fixed.